pyvista/core/objects.py

```diff
@@ -489,26 +705,104 @@ def cube_map(self, flag):
+
         Returns
         -------
         pyvista.Texture
             Copied texture.
         """
-        return Texture(self.to_image().copy())
+        return Texture(self.to_image().copy(deep=deep))
 
     def to_skybox(self):
         """Return the texture as a ``vtkSkybox`` if cube mapping is enabled.
 
         Returns
         -------
         vtk.vtkSkybox
-            Skybox if cube mapping is enabled.  Otherwise, ``None``.
+            Skybox if cube mapping is enabled.  Otherwise, raises an exception.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_sky_box_cube_map()  # doctest:+SKIP
+        >>> texture.to_skybox()  # doctest:+SKIP
+        <vtkmodules.vtkRenderingOpenGL2.vtkOpenGLSkybox(0x464dbb0) at 0x7f3130fab1c0>
 
         """
-        if self.cube_map:
-            skybox = _vtk.vtkSkybox()
-            skybox.SetTexture(self)
-            return skybox
+        if not self.cube_map:
+            raise ValueError('This texture is not a cube map.')
+        skybox = _vtk.vtkSkybox()
+        skybox.SetTexture(self)
+        return skybox
+
+    @property
+    def image_data(self):
+        """Return the underlying image data associated with this texture.
+
+        Returns
+        -------
+        pyvista.pyvista_ndarray
+            Array of the image data.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_masonry_texture()
+        >>> texture.image_data.shape
+        (256, 256, 3)
+        >>> texture.image_data.dtype
+        dtype('uint8')
+
+        """
+        data = pyvista.wrap(self.GetInput().GetPointData().GetScalars())
+        if data.ndim > 1:
+            shape = (self.dimensions[1], self.dimensions[0], data.shape[1])
+        else:
+            shape = (self.dimensions[1], self.dimensions[0])
+
+        return np.flip(data.reshape(shape), axis=1).swapaxes(1, 0)
+
+    @image_data.setter
+    def image_data(self, image_data):
+        """Set the image data."""
+        self._from_array(image_data)
+
+    def to_greyscale(self):
+        """Convert this texture as a single component (greyscale) texture.
+
+        Returns
+        -------
+        pyvista.Texture
+            Texture converted to greyscale. If already black and white,
+
+        Notes
+        -----
+        The transparency channel (if available) will be dropped.
+
+        Uses NTSC/PAL implementation.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> texture = examples.download_masonry_texture()
+        >>> bw_texture = texture.to_greyscale()
+        >>> bw_texture.plot()
+
+        """
+        if self.n_components == 1:
+            return self
+
+        new_texture = self.copy()
+        r, g, b = np.asarray(new_texture.image_data).T[:3]
+        greyscale = (0.2989 * r + 0.5870 * g + 0.1140 * b).astype(np.uint8)
```

Comment:
Checking every possible RGB triplet shows that truncation here from `astype` would make us lose the 255 value:
```py
>>> r, g, b = np.mgrid[:256, :256, :256]
>>> gray = 0.2989 * r + 0.5870 * g + 0.1140 * b
>>> gray.max()
254.97449999999998
>>> gray.astype(np.uint8).max()
254
>>> gray.round().astype(np.uint8).max()
255
```

So I recommend rounding here:
```suggestion
        greyscale = (0.2989 * r + 0.5870 * g + 0.1140 * b).round().astype(np.uint8)
```